### PR TITLE
feat(BottomSheet): add `zindex` prop & improve focus management logic

### PR DESCRIPTION
**Pull request by snitin315**
Fixes https://github.com/razorpay/blade/issues/1269

- Utilize [FloatingOverlay](https://floating-ui.com/docs/FloatingOverlay).
- Expose `zIndex` prop to BottomSheet

> **Note**
> I had to Skip the SSR test as it failed with error: [ReferenceError: navigator is not defined](https://github.com/floating-ui/floating-ui/blob/94ba5b7c7069c272c967e4eef6ba09a793880855/packages/utils/react/src/index.ts#L109) coming from `@floating-ui/react`, which is used in the BottomSheet component. Similarly,  Modal also uses @floating-ui/react and don't have SSR tests

### CODE REVIEW

File: packages/blade/src/components/BottomSheet/__tests__/BottomSheet.web.test.tsx [hunk 1]
```diff
@@ -205,34 +223,34 @@ describe('<BottomSheet />', () => {
         </Dropdown>
       );
     };
-    const { queryByTestId, getByRole } = renderWithTheme(<Example />);
+    const { queryByTestId, getByRole, getByLabelText } = renderWithTheme(<Example />);
 
     // open / close by clicking the select
     expect(queryByTestId('bottomsheet-body')).not.toBeVisible();
-    expect(getByRole('combobox', { name: 'Select Action' })).toBeInTheDocument();
```
Comment (anuraghazra) [inline packages/blade/src/components/BottomSheet/__tests__/BottomSheet.web.test.tsx:212]: Why did we removed getByRole?

File: .changeset/tricky-ducks-fetch.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@razorpay/blade": patch
+---
+
+feat(BottomSheet): add `zindex` prop & improve focus management logic
```
Comment (anuraghazra) [inline .changeset/tricky-ducks-fetch.md:5]: looks good, let's attribute @archie252000 for the initial draft.

File: packages/blade/src/components/BottomSheet/types.ts [hunk 3]
```diff
@@ -31,6 +31,11 @@ type BottomSheetProps = {
    * By default the initial focus will go to the close button
    */
   initialFocusRef?: React.MutableRefObject<any>;
+  /**
+   * Sets the z-index of the bottom sheet
```
[the remark this answers is not in the document]
Comment (snitin315) [reply packages/blade/src/components/BottomSheet/types.ts:35]: Done.

File: .changeset/tricky-ducks-fetch.md (hunk 2 above)
Comment (snitin315) [reply .changeset/tricky-ducks-fetch.md:5]: Done.

File: packages/blade/src/components/BottomSheet/__tests__/BottomSheet.ssr.test.tsx [hunk 4]
```diff
@@ -11,7 +11,10 @@ beforeAll(() => jest.spyOn(console, 'error').mockImplementation());
 afterAll(() => jest.restoreAllMocks());
 
 describe('<BottomSheet />', () => {
-  it('should render a BottomSheet ssr', () => {
+  // Skipping this test as it is fails in SSR with error: ReferenceError: navigator is not defined
+  // coming from @floating-ui/react, which is used in BottomSheet component
+  // https://github.com/floating-ui/floating-ui/blob/94ba5b7c7069c272c967e4eef6ba09a793880855/packages/utils/react/src/index.ts#L109
+  it.skip('should render a BottomSheet ssr', () => {
```
Comment (saurabhdaware) [inline packages/blade/src/components/BottomSheet/__tests__/BottomSheet.ssr.test.tsx:17]: is it specific to test or will bottomsheet break on actual SSR apps?

Comment (snitin315) [reply packages/blade/src/components/BottomSheet/__tests__/BottomSheet.ssr.test.tsx:17]: I tried it in a Next.js app, seems to be working.

https://github.com/razorpay/blade/assets/46647141/a89084d9-5356-4a2e-8586-ff84cb7683a6

**Review by saurabhdaware — APPROVED** [review]
Looks good to me

**Pull request merged**